x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -224,6 +237,11 @@ public synchronized void stop(ActionListener<StopDataFrameTransformAction.Respon
 
     @Override
     public synchronized void triggered(Event event) {
+        if (taskState.get() == DataFrameTransformTaskState.FAILED) {
```

Comment:
I moved this check up to this part of the call stack since now `maybeTriggerAsyncJob` is not able to be overridden.